DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h

```diff
@@ -0,0 +1,189 @@
+    
+        inline double getXPullNormalization() const { return residual_.first / pull_.first; }
+        inline double getYPullNormalization() const { return residual_.second / pull_.second; }
+
+        inline void setIsUsedForFit(bool usedForFit) {
+            if(usedForFit) isRealHit_ = true; 
+            isUsedForFit_ = usedForFit; 
+        }
+        inline bool getIsUsedForFit() const { return isUsedForFit_; }
+
+        inline void setIsRealHit(bool realHit) { 
+            if(!realHit) isUsedForFit_ = false;
+            isRealHit_ = realHit; 
+        }
+        inline bool getIsRealHit() const { return isRealHit_; }
+    
+      private:
+        TVector3 space_point_on_det_ ;  ///< mm
+        std::pair<double,double> residual_;  ///< mm
+        std::pair<double,double> pull_    ;  ///< normalised residual
+        bool isUsedForFit_;
+        bool isRealHit_;
+    };
+
+  public:
+    ///< parameter vector size
+    static const int dimension = 4;
+
+    ///< covariance matrix size
+    static const int covarianceSize = dimension * dimension;
+
+    CTPPSPixelLocalTrack() : z0_(0), chiSquared_(0), valid_(false), numberOfPointUsedForFit_(0)
+    {
+    }
+
+    CTPPSPixelLocalTrack(double z0, const TVectorD &track_params_vector,
+      const TMatrixD &par_covariance_matrix, double chiSquared);
+
+    virtual ~CTPPSPixelLocalTrack() {}
+
+    inline const edm::DetSetVector<CTPPSPixelFittedRecHit>& getHits() const { return track_hits_vector_; }
+    inline void addHit(unsigned int detId, const CTPPSPixelFittedRecHit &hit)
+    {
+      track_hits_vector_.find_or_insert(detId).push_back(hit);
+      if(hit.getIsUsedForFit()) ++numberOfPointUsedForFit_;
+    }
+
+    inline double getX0() const { return track_params_vector_[0]; }
+    inline double getX0Sigma() const { return sqrt(CovarianceMatrixElement(0, 0)); }
+    inline double getX0Variance() const { return CovarianceMatrixElement(0, 0); }
+
+    inline double getY0() const { return track_params_vector_[1]; }
+    inline double getY0Sigma() const { return sqrt(CovarianceMatrixElement(1, 1)); }
+    inline double getY0Variance() const { return CovarianceMatrixElement(1, 1); }
+
+    inline double getZ0() const { return z0_; }
+    inline void setZ0(double z0) { z0_ = z0; }
+
+    inline double getTx() const { return track_params_vector_[2]; }
+    inline double getTxSigma() const { return sqrt(CovarianceMatrixElement(2, 2)); }
+
+    inline double getTy() const { return track_params_vector_[3]; }
+    inline double getTySigma() const { return sqrt(CovarianceMatrixElement(3, 3)); }
+
+    inline TVector3 getDirectionVector() const
+    {
+      TVector3 vect(getTx(), getTy(), 1);
+      vect.SetMag(1.0);
+      return vect;
+    }
+
+    TVectorD getParameterVector() const;
+    void setParameterVector(const TVectorD & track_params_vector);
+
+    TMatrixD getCovarianceMatrix() const;
+    void setCovarianceMatrix(const TMatrixD &par_covariance_matrix);
```

Comment:
modifiers for params,  chi2, and cov are apparently not needed.
The functionality is provided by the constructor